### ISSUE TITLE
TestClientDisconnectRedundant is flaky

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#3350]
The following test appears to have failed:

[#9924](https://circleci.com/gh/cockroachdb/cockroach/9924):

```
W1207 22:14:18.992316 498 gossip/gossip.go:616  missing sentinel gossip; assuming partition and reconnecting
W1207 22:14:19.657257 498 gossip/gossip.go:616  missing sentinel gossip; assuming partition and reconnecting
W1207 22:14:19.893466 498 gossip/gossip.go:616  missing sentinel gossip; assuming partition and reconnecting
W1207 22:14:20.330565 498 rpc/server.go:436  rpc: write response failed: write tcp 127.0.0.1:50689->127.0.0.1:49493: use of closed network connection
W1207 22:14:20.330837 498 gossip/gossip.go:616  missing sentinel gossip; assuming partition and reconnecting
--- FAIL: TestClientDisconnectRedundant (10.12s)
	<autogenerated>:32: gossip/client_test.go:249: condition failed to evaluate within 10s: not connected
=== RUN   TestGossipInfoStore
W1207 22:14:20.334251 498 rpc/server.go:436  rpc: write response failed: write tcp 127.0.0.1:39531->127.0.0.1:54552: use of closed network connection
--- PASS: TestGossipInfoStore (0.00s)
=== RUN   TestGossipGetNextBootstrapAddress
I1207 22:14:20.337723 498 gossip/gossip_test.go:93  getting next address
I1207 22:14:20.337797 498 gossip/gossip_test.go:93  getting next address
I1207 22:14:20.337848 498 gossip/gossip_test.go:93  getting next address
I1207 22:14:20.337892 498 gossip/gossip_test.go:93  getting next address
I1207 22:14:20.337940 498 gossip/gossip_test.go:93  getting next address
--
W1207 22:14:21.036235 498 rpc/server.go:436  rpc: write response failed: write tcp 127.0.0.1:55021->127.0.0.1:33083: use of closed network connection
W1207 22:14:21.037079 498 rpc/server.go:436  rpc: write response failed: write tcp 127.0.0.1:56447->127.0.0.1:39062: use of closed network connection
W1207 22:14:21.037466 498 rpc/server.go:436  rpc: write response failed: write tcp 127.0.0.1:56447->127.0.0.1:39062: use of closed network connection
W1207 22:14:21.037918 498 rpc/server.go:436  rpc: write response failed: write tcp 127.0.0.1:56447->127.0.0.1:39064: use of closed network connection
--- PASS: TestConvergence (0.60s)
FAIL
FAIL	github.com/cockroachdb/cockroach/gossip	10.933s
=== RUN   TestParseResolverSpec
--- PASS: TestParseResolverSpec (0.00s)
=== RUN   TestGetAddress
--- PASS: TestGetAddress (0.00s)
PASS
ok  	github.com/cockroachdb/cockroach/gossip/resolver	1.018s
?   	github.com/cockroachdb/cockroach/gossip/simulation	[no test files]
=== RUN   TestKeySorting
--- PASS: TestKeySorting (0.00s)
=== RUN   TestMakeKey

```
Please assign, take a look and update the issue accordingly.

**Comment by tschottdorf** [comment]
repros easily with `make stressrace PKG=./gossip TESTS=ClientDisconnectRed`.

**Comment by tschottdorf** [comment]
fixed in #3378
